statsmodels/stats/multitest.py

```diff
@@ -444,3 +444,241 @@ def fdrcorrection_twostage(pvals, alpha=0.05, method='bky', iter=False,
+    sd0 : float
+        The estimated standard deviation of the empirical null distribution
+    prob0 : float
+        The estimated proportion of true null hypotheses
+
+    References
+    ----------
+    B Efron (2008).  Microarrays, Empirical Bayes, and the Two-Groups
+    Model.  Statistical Science 23:1, 1-22.
+
+    Notes
+    -----
+    See also:
+
+    http://nipy.org/nipy/labs/enn.html#nipy.algorithms.statistics.empirical_pvalue.NormalEmpiricalNull.fdr
+    """
+
+    def __init__(self, zscores, null_lb=-1, null_ub=1, estimate_mean=True,
+                 estimate_scale=True, estimate_prob=False):
+
+        # Extract the null z-scores
+        ii = np.flatnonzero((zscores >= null_lb) & (zscores <= null_ub))
+        if len(ii) == 0:
+            raise RunTimeError("No Z-scores fall between null_lb and null_ub")
+        zscores0 = zscores[ii]
+
+        # Number of Z-scores, and null Z-scores
+        n_zs, n_zs0 = len(zscores), len(zscores0)
+
+        # Unpack and transform the parameters to the natural scale, hold
+        # parameters fixed as specified.
+        def xform(params):
+
+            mean0 = 0.
+            sd0 = 1.
+            prob0 = 1.
+
+            ii = 0
+            if estimate_mean:
+                mean0 = params[ii]
+                ii += 1
+            if estimate_scale:
+                sd0 = np.exp(params[ii])
+                ii += 1
+            if estimate_prob:
+                prob0 = 1 / (1 + np.exp(-params[ii]))
+
+            return mean0, sd0, prob0
+
+
+        from scipy.stats.distributions import norm
+
+
+        def fun(params):
+            """
+            Negative log-likelihood of z-scores.
+
+            The function has three arguments, packed into a vector:
+
+            mean : location parameter
+            logscale : log of the scale parameter
+            logitprop : logit of the proportion of true nulls
+
+            The implementation follows section 4 from Efron 2008.
+            """
+
+            d0, s0, p0 = xform(params)
+
+            # Mass within the central region
+            central_mass = (norm.cdf((null_ub - d0) / s0) -
+                            norm.cdf((null_lb - d0) / s0))
+
+            # Probability that a Z-score is null and is in the central region
+            cp = p0 * central_mass
+
+            # Binomial term
+            rval = n_zs0 * np.log(cp) + (n_zs - n_zs0) * np.log(1 - cp)
+
+            # Truncated Gaussian term for null Z-scores
+            zv = (zscores0 - d0) / s0
+            rval += np.sum(-zv**2 / 2) - n_zs0 * np.log(s0)
+            rval -= n_zs0 * np.log(central_mass)
+
+            return -rval
+
+
+        # Estimate the parameters
+        from scipy.optimize import minimize
+        # starting values are mean = 0, scale = 1, p0 ~ 1
+        mz = minimize(fun, np.r_[0., 0, 3], method="Nelder-Mead")
+        mean0, sd0, prob0 = xform(mz['x'])
+
+        self.mean0 = mean0
+        self.sd0 = sd0
+        self.prob0 = prob0
```

Comment:
I think we should also here switch to an explicit `null`   (0 could just be a numbering, in general)
`mean_null` `sd_null` `prob_null` ?
or maybe without `0` and without `null` since the name already says it's the Null distribution, and we don't need to add any `_null` to pdf for consistency.